src/ConcordiumNetSdk/Client/Client.cs

```diff
@@ -0,0 +1,863 @@
+    /// </summary>
+    private readonly GrpcChannel _grpcChannel;
+
+    /// <summary>
+    /// The configuration to use with the client.
+    /// </summary>
+    private readonly ClientConfiguration _config;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Client"/> class.
+    /// </summary>
+    /// <param name="endpoint">
+    /// Endpoint of a resource where the V2 API is served. Any protocol scheme, port or query
+    /// parameters are ignored and the protocol defaults to either <c>http</c> or <c>https</c>
+    /// depending on whether a secure connection was specified in <paramref name="secure"/>.
+    /// </param>
+    /// <param name="port">
+    /// Port of the resource where the V2 API is served. This will override any port
+    /// specified in <paramref name="endpoint"/>.
+    /// </param>
+    /// <param name="timeout">The request timeout in seconds (default: <c>30</c>).</param>
+    /// <param name="secure">Flag indicating whether the client must use a secure connection (default: <c>true</c>).</param>
+    public Client(Uri endpoint, UInt16 port, ulong timeout = 30, bool secure = true)
+        : this(endpoint, port, new ClientConfiguration(timeout, secure)) { }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Client"/> class.
+    /// </summary>
+
+    /// <param name="endpoint">
+    /// Endpoint of a resource where the V2 API is served. Any protocol scheme, port or query
+    /// parameters are ignored and the protocol defaults to either <c>http</c> or <c>https</c>
+    /// depending on whether a secure connection was specified in <paramref name="configuration"/>.
+    /// </param>
+    /// <param name="port">
+    /// Port of the resource where the V2 API is served. This will override any port
+    /// specified in <paramref name="endpoint"/>.
+    /// </param>
+    /// <param name="configuration">The configuration to use with this client.</param>
+    public Client(Uri endpoint, UInt16 port, ClientConfiguration configuration)
+    {
+        GrpcChannelOptions options = new GrpcChannelOptions
+        {
+            Credentials = configuration.Secure
+                ? ChannelCredentials.SecureSsl
+                : ChannelCredentials.Insecure
+        };
+        _grpcChannel = GrpcChannel.ForAddress(
+            // The GRPC library expects a url with protocol that
+            // matches the intention of the <c>Secure</c> flag.
+            (configuration.Secure ? "https://" : "http://")
+                + endpoint.Host
+                + ":"
+                + port.ToString()
+                + endpoint.AbsolutePath,
+            options
+        );
+        InternalClient = new Queries.QueriesClient(_grpcChannel);
+        _config = configuration;
+    }
+
+    /// <summary>
+    /// Process a stream of blocks that arrive from the time the query is made onward.
+    /// This can be used to listen for incoming blocks. Note that this is non-terminating.
```

Comment:
Also make it clear that if the client is slow then there might be blocks that it does not see.

Reply:
Will do. Do the blocks also arrive in order of increasing height?

Reply:
Yes for finalized, not necessarily for the other one. However per-branch they arrive like that.